android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/Router.kt

```diff
@@ -34,7 +34,7 @@ protected constructor(
   public open val interactor: I,
   private val ribRefWatcher: RibRefWatcher,
   private val mainThread: Thread,
-) {
+) : RibComponent {
```

Comment:
Probably too generic for a name for this set of callbacks.  How about something more explicit around emitting callbacks around lifecycles, like RibEventEmitter?

Reply:
Good idea, I like this better

Reply:
Decided to go with `RibActionEmitter` to avoid confusion with existing `RibEventType`